Proofreading MO Guide (#11605)
openvinotoolkit/openvino: docs/MO_DG/prepare_model/convert_model/tf_specific/Convert_YOLO_From_Tensorflow.md

# Converting TensorFlow YOLO Models {#openvino_docs_MO_DG_prepare_model_convert_model_tf_specific_Convert_YOLO_From_Tensorflow}

This document explains how to convert real-time object detection YOLOv1, YOLOv2, YOLOv3 and YOLOv4 public models to the Intermediate Representation (IR). All YOLO models are originally implemented in the DarkNet framework and consist of two files:

> * *`.cfg`* file with model configurations

Reviewer: ```suggestion
* The *`.cfg`* file with model configurations
```

> * *`.weights`* file with model weights

Reviewer: ```suggestion
* The *`.weights`* file with model weights
```

mo --saved_model_dir yolov4 --output_dir models/IRs --input_shape [1,608,608,3] --model_name yolov4
```

> ## <a name="yolov3-to-ir"></a>Converting YOLOv3 Model to the IR

Reviewer 2: ```suggestion
## <a name="yolov3-to-ir"></a>Converting YOLOv3 Model to the OpenVINO format
```

On GitHub*, you can find several public versions of TensorFlow YOLOv3 model implementation. This section explains how to convert YOLOv3 model from

> the [https://github.com/mystic123/tensorflow-yolo-v3](https://github.com/mystic123/tensorflow-yolo-v3) repository (commit ed60b90) to an IR , but the process is similar for other versions of TensorFlow YOLOv3 model.

Author: Um, I am not sure I follow your thought.

> the [https://github.com/mystic123/tensorflow-yolo-v3](https://github.com/mystic123/tensorflow-yolo-v3) repository (commit ed60b90) to an IR , but the process is similar for other versions of TensorFlow YOLOv3 model.

Reviewer: ```suggestion
the [repository) (commit ed60b90) to an IR , but the process is similar for other versions of TensorFlow YOLOv3 model.
```
Left second repository by accident.

> the [https://github.com/mystic123/tensorflow-yolo-v3](https://github.com/mystic123/tensorflow-yolo-v3) repository (commit ed60b90) to an IR , but the process is similar for other versions of TensorFlow YOLOv3 model.

Reviewer: ```suggestion
the [repository](https://github.com/mystic123/tensorflow-yolo-v3) (commit ed60b90) to an IR , but the process is similar for other versions of TensorFlow YOLOv3 model.
```

> the [https://github.com/mystic123/tensorflow-yolo-v3](https://github.com/mystic123/tensorflow-yolo-v3) repository (commit ed60b90) to an IR , but the process is similar for other versions of TensorFlow YOLOv3 model.

Author: ```suggestion
the [repository](https://github.com/mystic123/tensorflow-yolo-v3) (commit ed60b90) to an IR , but the process is similar for other versions of TensorFlow YOLOv3 model.
```

git checkout ed60b90
```
3. Download [coco.names](https://github.com/AlexeyAB/darknet/blob/master/data/coco.names) file from the DarkNet website **OR** use labels that fit your task.

> 4. Download the [yolov3.weights](https://pjreddie.com/media/files/yolov3.weights) (for the YOLOv3 model) or [yolov3-tiny.weights](https://pjreddie.com/media/files/yolov3-tiny.weights) (for the YOLOv3-tiny model) file **OR** use your pretrained weights with the same structure.

Reviewer 2: ```suggestion
4. Download the [yolov3.weights](https://pjreddie.com/media/files/yolov3.weights) (for the YOLOv3 model) or [yolov3-tiny.weights](https://pjreddie.com/media/files/yolov3-tiny.weights) (for the YOLOv3-tiny model) file **OR** use your pre-trained weights with the same structure.
```

```sh
python3 convert_weights_pb.py --class_names coco.names --data_format NHWC --weights_file yolov3-tiny.weights --tiny
```
At this step, you may receive a warning like *`WARNING:tensorflow:Entity <...> could not be transformed and will be executed as-is.`*. To work around this issue, switch to gast 0.2.2 with the following command:
```sh
pip3 install --user gast==0.2.2
```

If you have YOLOv3 weights trained for an input image with the size different from 416 (320, 608 or your own), provide the *`--size`* key with the size of your image specified while running the converter. For example, run the following command for an image with size 608:
```sh
python3 convert_weights_pb.py --class_names coco.names --data_format NHWC --weights_file yolov3_608.weights --size 608
```

> ### Converting a YOLOv3 TensorFlow Model to the IR

Reviewer 2: ```suggestion
### Converting a YOLOv3 TensorFlow Model to the OpenVINO format
```

```sh
python3 flow --model <path_to_model>/<model_name>.cfg --load <path_to_model>/<model_name>.weights --labels <path_to_dataset_labels_file> --savepb
```

> For YOLOv1  the argument *`--labels`* can be skipped. If the model was successfully converted, you can find the *`<model_name>.meta`* and *`<model_name>.pb`* files

Author: ```suggestion
For YOLOv1,  the *`--labels`* argument can be skipped. If the model was successfully converted, you can find the *`<model_name>.meta`* and *`<model_name>.pb`* files.
```
